client/xdag_config.c

```diff
@@ -0,0 +1,536 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+//#include <errno.h>
+#include "xdag_config.h"
+
+
+
+#ifndef true
+#define true (1)
+#endif
+
+#ifndef false
+#define false (0)
+#endif
+
+#if defined (WIN32)
+#define strkscmp stricmp
+#else
+#define strkscmp strcasecmp
+#endif
+
+static char *trim(char *str)
+{
+    if(str != NULL)
+    {
+        int len = (int)strlen(str);
+        char *start = str;
+        char *end = str + len;
+        while(start < end && *start <= ' ')
+        {
+            start++;
+        }
+        while(end > start && *(end-1) <= ' ')
+        {
+            end--;
+        }
+        if(end - start != len)
+        {
+            strncpy(str, start, end-start);
+            str[end-start] = '\0';
+        }
+    }
+    return str;
+}
+
+
+
+
+
+typedef struct
+{
+    char *name;
+    char *value;
+} KEY;
+
+typedef struct
+{
+    char *name;
+    int number_keys;
+    KEY **keys;
+    char **key_names;
+} NODE;
+
+
+
+typedef struct
+{
+    char *path;
+    int number_nodes;
+    NODE **nodes;
+    char **node_names;
+} CFG;
+
+
+static const char COMMENT_MARK[] = {'#', '*'};
+typedef enum
+{
+    TYPE_EMPTYLINE,
+	TYPE_COMMENT,
+    TYPE_NODE,
+    TYPE_KEY,
+    TYPE_UNKNOW,
+} CFG_CONTENT_TYPE;
+
+static char *readline(FILE *fp)
+{
+    char buf[128];
+    char c;
+    char *line = NULL;
+    int line_len = 0, buf_len = 0;
+
+    while(!feof(fp))
```

Comment:
You read line from file. Do the lines really can be so long, so you need to read them in several iterations?

Reply:
Because I am worried that there may be an expansion later.